utils/tini/Makefile

```diff
@@ -0,0 +1,44 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=tini
+PKG_VERSION:=0.18.0
+PKG_RELEASE:=1
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/krallin/tini/tar.gz/v${PKG_VERSION}?
+PKG_HASH:=1097675352d6317b547e73f9dc7c6839fd0bb0d96dafc2e5c95506bb324049a2
+
+PKG_MAINTAINER:=Gerard Ryan <G.M0N3Y.2503@gmail.com>
+
+CMAKE_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/tini
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=simplest init utility
+  URL:=https://github.com/krallin/tini
+  DEPENDS:=
+endef
+
+define Package/tini/description
+A tiny but valid init process for containers.
+endef
+
+TARGET_CFLAGS+= \
+  -DPR_SET_CHILD_SUBREAPER=36 \
+  -DPR_GET_CHILD_SUBREAPER=37
+
+define Package/tini/install
+	$(INSTALL_DIR) $(1)/usr/bin
+
+	# static version seemes to be effected by https://www.openwall.com/lists/musl/2018/07/18/8 so disabled for now
+	#$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/tini-static $(1)/usr/bin/
```

Comment:
Is this issue present in current OpenWrt toolchain?  It seems that they have figured out the cause and provided some workaround.  It seems to me tini depending on host libc is very limited in usefulness.

Reply:
I compiled with symbols and came across a similar if not the same error in the same area of code.
It also looks like OpenWrt uses the same [version of Binutils](https://github.com/openwrt/openwrt/blob/8568dcd931c53ba177f1c9601e9706d95d015062/toolchain/binutils/Config.in) as mentioned in the linked mailing list. I figured that from my testing the dynamic version worked with docker so we could just use that one until Binutils is updated.
Also, I'm not quite sure what you mean about the host libc dependency.

Reply:
> Also, I'm not quite sure what you mean about the host libc dependency.

By host I mean the target system tini will load and run.  If I understand it correctly, dockerd will copy tini inside the container and load there.  If tini depends on loader from musl-libc and if the container was not musl-libc-based, then very likely the following error will appear.


```
standard_init_linux.go:207: exec user process caused "no such file or directory"
```

It's very important tini be a static binary.

Reply:
I am wondering whether we can workaround this by tweaking linker flags like said in the musl mailing list.

Backporting that fix from binutils is also an option.  But I think it requires time to do it right.

Reply:
Right, it didn't occur to me that it was copying in tini, I was under the impression that it was just part of the init process in docker that requires libc, Good catch! 
I added the workaround for tini and all seems to be working fine in non-libc images too.